core/src/Cuda/Kokkos_Cuda_Team.hpp

```diff
@@ -161,8 +161,34 @@ class CudaTeamMember {
   //--------------------------------------------------------------------------
 
   template <class ValueType>
-  KOKKOS_INLINE_FUNCTION void team_broadcast(ValueType& val,
-                                             const int& thread_id) const {
+  KOKKOS_INLINE_FUNCTION
+      typename std::enable_if<(sizeof(ValueType) < 4), void>::type
+      team_broadcast(ValueType& val, const int& thread_id) const {
+    using conv_type = uint32_t;
```

Comment:
Why did you chose this over `int`?

Reply:
[Minimum width of `int` is 16 bits](https://eel.is/c++draft/basic.fundamental) btw.

Reply:
My point was `in_place_shfl` has support for `int`.

Reply:
we draw the line at 4 bytes (in the template specialization), so we need to use a 4 byte type

Reply:
This should be inside the `#ifdef __CUDA_ARCH__`.